File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsInputMedia.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.core.TLObject;

/**
 * Abstraction level for the following constructors:
 * <ul>
 * <li>{@link TLInputMediaContact}: inputMediaContact#a6e45987</li>
 * <li>{@link TLInputMediaDocument}: inputMediaDocument#1a77f29c</li>
 * <li>{@link TLInputMediaDocumentExternal}: inputMediaDocumentExternal#e5e9607c</li>
 * <li>{@link TLInputMediaEmpty}: inputMediaEmpty#9664f57f</li>
 * <li>{@link TLInputMediaGame}: inputMediaGame#d33f43f3</li>
 * <li>{@link TLInputMediaGeoPoint}: inputMediaGeoPoint#f9c44144</li>
 * <li>{@link TLInputMediaGifExternal}: inputMediaGifExternal#4843b0fd</li>
 * <li>{@link TLInputMediaInvoice}: inputMediaInvoice#92153685</li>
 * <li>{@link TLInputMediaPhoto}: inputMediaPhoto#e9bfb4f3</li>
 * <li>{@link TLInputMediaPhotoExternal}: inputMediaPhotoExternal#b55f4f18</li>
 * <li>{@link TLInputMediaUploadedDocument}: inputMediaUploadedDocument#d070f1e9</li>
 * <li>{@link TLInputMediaUploadedPhoto}: inputMediaUploadedPhoto#630c9af1</li>
 * <li>{@link TLInputMediaUploadedThumbDocument}: inputMediaUploadedThumbDocument#50d88cae</li>
 * <li>{@link TLInputMediaVenue}: inputMediaVenue#2827a81a</li>
 * </ul>
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsInputMedia extends TLObject {

    public TLAbsInputMedia() {
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsInputMedia.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.core.TLObject

/**
 * Abstraction level for the following constructors:
 * * [inputMediaContact#f8ab7dfb][TLInputMediaContact]
 * * [inputMediaDocument#23ab23d2][TLInputMediaDocument]
 * * [inputMediaDocumentExternal#fb52dc99][TLInputMediaDocumentExternal]
 * * [inputMediaEmpty#9664f57f][TLInputMediaEmpty]
 * * [inputMediaGame#d33f43f3][TLInputMediaGame]
 * * [inputMediaGeoLive#7b1a118f][TLInputMediaGeoLive]
 * * [inputMediaGeoPoint#f9c44144][TLInputMediaGeoPoint]
 * * [inputMediaGifExternal#4843b0fd][TLInputMediaGifExternal]
 * * [inputMediaInvoice#f4e096c3][TLInputMediaInvoice]
 * * [inputMediaPhoto#b3ba0635][TLInputMediaPhoto]
 * * [inputMediaPhotoExternal#e5bbfe1a][TLInputMediaPhotoExternal]
 * * [inputMediaUploadedDocument#5b38c6c1][TLInputMediaUploadedDocument]
 * * [inputMediaUploadedPhoto#1e287d04][TLInputMediaUploadedPhoto]
 * * [inputMediaVenue#c13d1c11][TLInputMediaVenue]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsInputMedia : TLObject()
